lib/ui/window/platform_message_response_dart.cc

```diff
@@ -63,16 +73,44 @@ PlatformMessageResponseDart::~PlatformMessageResponseDart() {
 }
 
 void PlatformMessageResponseDart::Complete(std::unique_ptr<fml::Mapping> data) {
-  PostCompletion(std::move(callback_), ui_task_runner_, &is_complete_, channel_,
-                 [data = std::move(data)] {
-                   return tonic::DartByteData::Create(data->GetMapping(),
-                                                      data->GetSize());
-                 });
+  if (send_port_ == ILLEGAL_PORT) {
+    PostCompletion(std::move(callback_), ui_task_runner_, &is_complete_,
+                   channel_, [data = std::move(data)] {
+                     return tonic::DartByteData::Create(data->GetMapping(),
+                                                        data->GetSize());
+                   });
+  } else {
+    // TODO(gaaclarke): Move all this logic into its own
+    // PlatformMessageResponse.
+    is_complete_ = true;
+    Dart_CObject response = {
+        .type = Dart_CObject_kExternalTypedData,
+    };
+    uint8_t* copy = static_cast<uint8_t*>(malloc(data->GetSize() + 8));
+    memcpy(copy, &identifier_, sizeof(int64_t));
+    memcpy(copy + 8, data->GetMapping(), data->GetSize());
+    response.value.as_external_typed_data.type = Dart_TypedData_kByteData;
+    response.value.as_external_typed_data.length = data->GetSize();
+    response.value.as_external_typed_data.data = copy;
+    response.value.as_external_typed_data.peer = nullptr;
+    response.value.as_external_typed_data.callback = FreeFinalizer;
+    bool did_send = Dart_PostCObject(send_port_, &response);
```

Comment:
@mkustermann FYI this is what I ended up with to make the response.  I haven't cleaned it up or tested it.  It's just the best way I could think to send the data in one message.  The completers are stored in a `Map<int, Completer<ByteData?>>` in the framework pr.

Reply:
Alternatively (to avoid the copy - see also other comment), you can do this:
```c
Dart_CObject identifier;
identifier.type = Dart_CObject_kInt64;
identifier.value.as_int64 = identifier;

Dart_CObject bytes;
bytes.type = Dart_CObject_kExternalTypedData;
bytes.value.as_external_typed_data.... = ...;
bytes...

Dart_CObject* tuple[2] = {&identifier, &bytes};

Dart_CObject array;
array.kind = Dart_CObject_kArray;
array.value.as_array.length = 2;
array.value.as_array.values = tuple;

bool did_send = Dart_PostCObject(send_port_, &array);
```

What is identifier in this case?

Reply:
The identifier is an enumeration for each message that can be used to look up a Completer on the dart side.  I did what you suggested in the doc where I avoided sending a closure to the C side.  It looks like sending and maintaining an object owned by a background isolate will be a pain since I'd have to release the closure on the thread associated with the arbitrary isolate.  So, it was easier just to store a primitive datatype.

I looked at `Dart_CObject_kArray` but from the documentation it isn't clear to me how it manifests in dart since there is no documented type for it in [`typed_data`](https://api.dart.dev/be/181224/dart-typed_data/dart-typed_data-library.html).